ansible/roles/jenkins-worker/tasks/partials/tap2junit/ubuntu.yml

```diff
@@ -1,11 +1,28 @@
 ---
 
 #
-# ubuntu series: python 2.7
+# ubuntu series
 #
 
 - name: install pip
-  package: name=python-pip state=present
+  when: os != "ubuntu1604"
+  package: name=python3-pip state=present
+
+# RHEL 7 doesn't have a package for pip and EPEL 7 doesn't support s390x.
+# Install pip via `get-pip.py`.
+- name: ubuntu1604 | download pip install script
+  when: os == "ubuntu1604"
```

Comment:
I'm not sure any of the ubuntu1604 machines are connected to the test CI as they are still running Java 8. I'd prefer to orderly remove them (first from Jenkins, then Ansible) and then replace them with something newer (e.g. Ubuntu 20.04 or 22.04).